components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -1534,4 +1486,139 @@ private boolean isDateTimeAndMSSql(String attributeName) throws OrganizationMana
         return (VIEW_CREATED_TIME_COLUMN.equals(attributeName) || VIEW_LAST_MODIFIED_COLUMN.equals(attributeName))
                 && isMSSqlDB();
     }
+
+    private FilterQueryBuilder buildFilterQuery(List<ExpressionNode> expressionNodes)
+            throws OrganizationManagementServerException {
+
+        FilterQueryBuilder filterQueryBuilder = new FilterQueryBuilder();
+        appendFilterQuery(expressionNodes, filterQueryBuilder);
+        return filterQueryBuilder;
+    }
+
+    private FilterQueryBuilder buildParentIdFilterQuery(List<ExpressionNode> parentIdExpressionNodes) {
+
+        FilterQueryBuilder parentIdFilterQueryBuilder = new FilterQueryBuilder();
+        appendFilterQueryForParentId(parentIdFilterQueryBuilder, parentIdExpressionNodes);
+        return parentIdFilterQueryBuilder;
+    }
+
+    private String prepareGetOrganizationQuery(boolean authorizedSubOrgsOnly, boolean recursive, String sortOrder,
+                                               String applicationAudience, FilterQueryBuilder filterQueryBuilder,
+                                               FilterQueryBuilder parentIdFilterQueryBuilder, String userID)
+            throws OrganizationManagementServerException {
+
+        String sqlStmt = prepareSqlStatement(authorizedSubOrgsOnly, applicationAudience);
+        String getOrgSqlStmtTail = getOrgSqlStmtTail(authorizedSubOrgsOnly, applicationAudience);
+        if (filterQueryBuilder.getHasSubAttribute()) {
+            sqlStmt = sqlStmt.replace("WHERE", INNER_JOIN_UM_ORG_ATTRIBUTE);
+        }
+        sqlStmt = appendFilterQueries(sqlStmt, filterQueryBuilder, parentIdFilterQueryBuilder, getOrgSqlStmtTail,
+                recursive, sortOrder);
+
+        String username = getAuthenticatedUsername();
+        if (StringUtils.isNotEmpty(username)) {
+            username = UserCoreUtil.removeDomainFromName(username);
+        }
+
+        /* The shared user parent user might be created with user ID if there is business user with same name
+        in the child organization. */
+        return sqlStmt.replace(USER_NAME_LIST_PLACEHOLDER, Stream.of(username, userID)
+                .map(name -> "'" + name + "'").collect(Collectors.joining(",")));
+    }
+
+    private String prepareSqlStatement(boolean authorizedSubOrgsOnly, String applicationAudience) {
```

Comment:
This may not required to take as a private method as this is very specific logic which is highly unlikely reusable.
Break the complex methods into private methods is good but we can keep some limit other wise readability will decrease.

Reply:
Saw couple of other places also. Check with Anuradha on what should be the best practice.

Reply:
We decided to keep the private method but to rename the method to`getOrgSqlStatement` to increase readability.